pages/event-types/[type].tsx

```diff
@@ -69,7 +72,17 @@ const PERIOD_TYPES = [
 ];
 
 const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
-  const { eventType, locationOptions, availability, team, teamMembers } = props;
+  const { eventType, locationOptions, availability, team, teamMembers, localeProp } = props;
+  let locale = "en";
+  if (localeProp) {
+    locale = localeProp;
+  }
+
+  const { i18n } = useTranslation("event-type-edit-page");
+
+  useEffect(() => {
+    (async () => await i18n.changeLanguage(locale))();
+  }, [i18n, locale]);
```

Comment:
Is this the recommended way? Seems a bit flaky

Reply:
I think we can abstract this even further

Reply:
agree. let's abstract this